### DISCUSSION TITLE
Cline became unusable with new streaming feature

### DISCUSSION THREAD

**Discussion by AIHypnosis** [cline/cline#576] [Feature Requests]
Claude became Cline and became unusable! 
The new feature when response streamed with a yellow text decoration animation is very confusing and time consuming.
Even in a small file of about 300 lines you have to wait forever for this feature to complete scrolling. The problem is that the bot scrolls through the whole file, line by line, not just the block of code that needs to be changed. Also, this makes it impossible to undo the change if something went wrong with the code. This feature is a move in the wrong direction! Please revert this feature back asap. 
Alternatively, you can make it possible to disable this feature in settings.

Thanks!

**Comment by tsdevau** [comment]
And here I felt I was all alone on this issue.

I find it tedious and of no value what-so-ever.  It actually feels slower than just waiting for the completed diff to appear.  Unfortunately, I think everyone has become use to this streaming approach with LLMs.  So it feels like it's a _"feature"_ that's here to stay.

I too just hope that some consideration is given to adding an option to disable streaming as a user preference.

**Comment by heltonteixeira** [comment]
The time to show a full file in both the stream or previous format is the same.

You can undo or revert any changes in the files using the built-in timeline feature of VSCode.

**Reply by AIHypnosis** [reply]
What do you mean the time to show a full file in both the stream or previous format is the same? It is not. It is much slower. 
For 350 lines of code file I have to wait forever now for Cline to scroll through all the lines! Before, the result was almost immediate. 

 Can you collaborate more on that:
 "You can undo or revert any changes in the files using the built-in timeline feature of VSCode"
 I am not familiar with this feature. Where is it in VScode?

**Reply by tsdevau** [reply]
The actual speed or time may be the same, but the perception of it is definitely not.

From my standpoint, the time is a moot point anyway.  My issues with the feature are primarily around:
  - It's insanely distracting and painful to watch
  - The stream offers no benefit
  
If I could stop following the stream and stay at the top of the file to start reviewing the diff, then, yes, there would be a benefit to the streamed response.  This would enable you to begin reviewing the diff as it arrives.

Alas, as it stands, if you try to scroll back to a change above the current progress, you're immediately dragged back down to the next line of progress.  Thereby rendering the stream worthless, as nothing can be done until it's finished.

**Comment by seoeaa** [comment]
I like everything

**Comment by AIHypnosis** [comment]
@heltonteixeira 
Can you collaborate more on that please:
""You can undo or revert any changes in the files using the built-in timeline feature of VSCode"
I am not familiar with this feature and I could not find it in VScode?

**Reply by tsdevau** [reply]
@AIHypnosis the _TIMELINE_ is a section usually in the Explorer sidebar.  If you can't see the section, click the `...` at the top right of the Explorer sidebar and make sure there is a tick against 'Timeline'.

When you expand the timeline section you can see the save and commit history of the file currently active in the editor window.  From this section you can revert to a point in the timeline of the file.

It's pretty easy to find, but if you can't locate it, I can add a screenshot of where to look.

---

That said, it's far easier and more practical to review the AI changes in the presented diff.  In which you can revert changed lines or sections independently before saving the changes.  This is where you should be reviewing what the AI suggests and reverting or editing as required.

If Cline's suggestions are just stupid or too many unrelated changes to manually revert, you can always reject his changes completely with the Reject button and have him have another go.

**Comment by blistovmhz** [comment]
I suspect the "feature" is not the "bug" we're probably talking about.

EG: I ask Cline to make some edit to a file of around 2000 lines. It has already determined the changes we're going to implement and has already shown me the suggested updates. I say go ahead and it starts scrolling forever, incredibly slowly (maybe 2-3 lines per second max) and errors out a few hundred lines in. NOthing ever completes. Just ate a freedom-buck trying to get it to add it's 12 lines of code to the file.

Wait, is this normal, expected behaviour? It's supposed to hijack your editor for 10 minutes at a time and force you to scroll through the entire file to find out it's errored out, despite the fact that it already read the file, already provided the correct code in the chat, but now .... I don't get it. What is the feature? Genuine question. FIrst time trying Cline. It hasn't yet successfully added a console log of "hello world" to the bottom of my file because it never gets that far.

**Reply by dantess** [reply]
same here

**Reply by MBrekhof** [reply]
https://github.com/microsoft/vscode/issues/236877 is an issue with VS Code becoming very slow. As Cline is just an extension could it be that the main cause could be VS Code?

**Comment by dikkietrom** [comment]
things like this make it defacto useless for long files. it seems the developer is not developing cline with cline, which is ironic. I try to fix cline with cline but the Cline.ts file is 3000 lines long and even when you want to refactor it is undoable so I quit trying to solve it. It is in the decoration class where it happens and i often dont even want the editor to open which leads to a tsunami of open files. I think when the developer starts using cline to develop cline things will change in this regard because you cannot develop cline with cline in it is current state. File should not be longer than 100 lines then it works reasonably ok, although i would prefer that opening an editor is an option. You can actually remove the tools and make a bash mcp that does it all and write in the sys prompt that it should not use these tools write_to_file , replace_in_file 

btw it is really easy to work on cline with cline, just pull it from git and work in the debug window. it is much better in explaining and analysing than coding anyways.

**Comment by blistovmhz** [comment]
Yea, it's absolutely brutal for anything over 100 lines. 3 full minutes do
apply a console log to a 1500 line file.

**Comment by dikkietrom** [comment]
the solution is simple i think, i switched to cursor but that sucks also, i want something that has all of these 2 and i am afraid i have to make it myself, cursor does double buffering, first writes it full in a buffer and then ask to apply and bam there is the diff, the cline dev should really make their stuff stable before hunting the next feature or there will be someone that forks it and does that, might be me

**Comment by dikkietrom** [comment]
epiphany : work on cline with cursor, as long as that is needed cursor wins

having cline plugged into cursor is already the way forward, have best of both but debug cline in cursor is new hope haha

<img width="1920" alt="image" src="https://github.com/user-attachments/assets/9737b59a-df8b-4583-a0ec-4fdf01cf47f9" />
